bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2641,6 +2643,8 @@ void ReplicaImp::onNewView(const std::vector<PrePrepareMsg *> &prePreparesForNew
   LOG_INFO(VC_LOG, "Start working in new view: " << KVLOG(curView));
 
   controller->onNewView(curView, primaryLastUsedSeqNum);
+  // pubish clients public keys, in case prev primar failed to publish
+  if (primaryIsMe) KeyExchangeManager::instance().publishClientsPublicKeys(SigManager::instance()->ClientsPublicKeys());
```

Comment:
@HristoStaykov can you confirm this code calls ```publishClientsPublicKeys``` on VC complete ?

Reply:
This is the point in the View Change process where the beginning of the rebuilding of the previous View's Working Window begins. After this point we have 300 Sequence ID-s for the Replicas to reach consensus on (as for now), but I think posting the Client Operation here should be ok. The Primary never complains against himself, and he is always the first to enter the View.